railties/lib/rails/tasks/routes.rake

```diff
@@ -1,7 +1,36 @@
-desc 'Print out all defined routes in match order, with names. Target specific controller with CONTROLLER=x.'
+require 'active_support/deprecation'
+require 'active_support/core_ext/string/strip' # for strip_heredoc
+require 'optparse'
+
+desc 'Print out all defined routes in match order, with names. Target specific controller with --controller option - or its -c shorthand.'
 task routes: :environment do
   all_routes = Rails.application.routes.routes
   require 'action_dispatch/routing/inspector'
   inspector = ActionDispatch::Routing::RoutesInspector.new(all_routes)
-  puts inspector.format(ActionDispatch::Routing::ConsoleFormatter.new, ENV['CONTROLLER'])
+  if ARGV.any?{|argv| argv.start_with? 'CONTROLLER'}
```

Comment:
Still needs to be `ARGV.any? { |argv| argv.start_with? 'CONTROLLER' }` with spaces 'round the braces 😁